pyaedt/icepak.py

```diff
@@ -3182,3 +3182,446 @@ def get_face_normal(obj_face):
+        props["Thickness"] = (thickness,)
+        props["Solid Material"] = material
+        props["External Condition"] = ext_condition
+        props["Heat Flux"] = heat_flux
+        props["Temperature"] = temperature
+        if htc_dataset is None:
+            props["Heat Transfer Coefficient"] = htc
+        else:
+            props["Heat Transfer Coefficient Variation Data"] = {
+                "Variation Type": "Temp Dep",
+                "Variation Function": "Piecewise Linear",
+                "Variation Value": '["1w_per_m2kel", "pwl({},Temp)"]'.format(htc_dataset),
+            }
+        props["Reference Temperature"] = ref_temperature
+        props["Heat Transfer Data"] = {
+            "Heat Transfer Correlation": ht_correlation,
+            "Heat Transfer Convection Type": "Forced Convection",
+        }
+        if ht_correlation:
+            props["Heat Transfer Data"].update(
+                {
+                    "Heat Transfer Correlation": True,
+                    "Heat Transfer Convection Type": ht_correlation_type,
+                    "Heat Transfer Convection Fluid Material": ht_correlation_fluid,
+                }
+            )
+            if ht_correlation_type == "Forced Convection":
+                props["Heat Transfer Data"].update(
+                    {
+                        "Flow Type": ht_correlation_flow_type,
+                        "Flow Direction": ht_correlation_flow_direction,
+                        "Heat Transfer Coeff Value Type": ht_correlation_value_type,
+                        "Stream Velocity": ht_correlation_free_stream_velocity,
+                    }
+                )
+            elif ht_correlation_type == "Natural Convection":
+                props["Heat Transfer Data"].update(
+                    {"Surface": ht_correlation_surface, "Ambient Temperature": ht_correlation_amb_temperature}
+                )
+        props["Radiation"] = {"Radiate": radiate, "RadiateTo": "AllObjects", "Surface Material": radiate_surf_mat}
+        props["Shell Conduction"] = shell_conduction
+        props["External Surface Radiation"] = ext_surf_rad
+        props["External Material"] = ext_surf_rad_material
+        props["External Radiation Reference Temperature"] = ext_surf_rad_ref_temp
+        props["External Radiation View Factor"] = ext_surf_rad_view_factor
+        bound = BoundaryObject(self, name, props, "Stationary Wall")
+        if bound.create():
+            self.boundaries.append(bound)
+        return bound
+
+    @pyaedt_function_handler()
+    def assign_stationary_wall_with_heat_flux(
+        self,
+        geometry,
+        name=None,
+        heat_flux="0irrad_W_per_m2",
+        thickness="0mm",
+        material="Al-Extruded",
+        radiate=False,
+        radiate_surf_mat="Steel-oxidised-surface",
+        shell_conduction=False,
+    ):
+        """Function to assign surface wall boundary condition with heat flux thermal assignment.
```

Comment:
```suggestion
        """Assign a surface wall boundary condition with heat flux thermal assignment.
```
Can the description be made more clear? I find it confusing that you are assigning a condition to an assignment.

Reply:
In all these descriptions, I'll use:

"Assign a surface wall boundary condition with specified ......"

This should be clearer.